shotover-proxy/tests/cassandra_int_tests/timestamp.rs

```diff
@@ -0,0 +1,93 @@
+use crate::helpers::cassandra::{assert_query_result, run_query, CassandraConnection, ResultValue};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+async fn flag(connection: &CassandraConnection) {
+    let timestamp = setup(connection).await;
+
+    let _r = connection
+        .execute_with_timestamp(
+            "UPDATE test_timestamps.test_table SET a = 'a1-modified-1' WHERE id = 0;",
+            timestamp,
+        )
+        .await;
+
+    assert_query_result(
+        connection,
+        "SELECT WRITETIME(a) FROM test_timestamps.test_table WHERE id = 0;",
+        &[&[ResultValue::BigInt(timestamp)]],
+    )
+    .await;
+}
+
+async fn query(connection: &CassandraConnection) {
+    let timestamp = setup(connection).await;
+
+    run_query(
+        connection,
+        &format!(
+            "UPDATE test_timestamps.test_table USING TIMESTAMP {} SET a = 'a1-modified-1' WHERE id = 0;",
+            timestamp
+        ),
+    )
+    .await;
+
+    assert_query_result(
+        connection,
+        "SELECT WRITETIME(a) FROM test_timestamps.test_table WHERE id = 0;",
+        &[&[ResultValue::BigInt(timestamp)]],
+    )
+    .await;
+}
+
+async fn setup(connection: &CassandraConnection) -> i64 {
```

Comment:
I dont really understand why we use this to generate a timestamp rather than just hardcoding a timestamp?

Reply:
Because if we send one with an earlier timestamp than Cassandra has as the last write (the insert) it will be ignored.

Reply:
Oooh, thats a good reason to do it like this, can you add a comment?

Also worth considering if we can just use the current system time, the only reason I think that could go wrong is if there is a risk of it ending up in what cassandra considers the future.

Lets also rename `setup` -> `get_timestamp`

Reply:
Changes look good, my only remaining question is why not just use the current system time?